File: 522 Longest Uncommon Subsequence II.kt
class Solution {
    fun findLUSlength(strs: Array<String>): Int {
        var len = -1
        for (i in 0 until strs.size) {
            var satisfied = true
            for (j in 0 until strs.size) {
                if (i == j) continue
                if (strs[i].isSubSeq(strs[j])) satisfied = false
            }
            if (satisfied) len = Math.max(len, strs[i].length)
        }
        return len
    }

    private fun String.isSubSeq(other: String): Boolean {
        if (this.length > other.length) return false
        var p = 0
        var q = 0
        while (p < this.length && q < other.length) {
            if (this[p] == other[q]) {
                p++
                q++
            } else {
                q++
            }
        }
        return p == this.length
    }
}

File: 522 Longest Uncommon Subsequence II.c

int max(int a,int b)
{
    return a>b ? a:b;
}


bool LCS(char* a, char* b) {
        if (strlen(a)<strlen(b)) return false;
        int i = 0;
        for(int k=0;k<strlen(a);k++) {
            if(i < strlen(b) && b[i] == a[k]) {
                i++;
            }
        }
        return i == strlen(b);
}


int findLUSlength(char ** strs, int strsSize){
    int ret=-1,i,j;

    for(i=0;i<strsSize;i++) {
        for(j=0;j<strsSize;j++){
            if(i==j)
                continue;
            if(LCS(strs[j],strs[i])){
                break;
            }
        }
        if(j==strsSize){
            ret=max(ret,strlen(strs[i]));
        }
    }
    return ret;
}
